In [2]:
import pandas as pd

In [45]:
#CARGAMOS NUESTRO ARCHIVO CON LOS DATOS
df_viviendas_california = pd.read_csv('housing.csv')

In [46]:
#DAREMOS UNA PRIMERA OBSERVACION SOBRE LA INFORMACION DE LOS METADATOS
df_viviendas_california.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 10 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   longitude           20640 non-null  float64
 1   latitude            20640 non-null  float64
 2   housing_median_age  20640 non-null  float64
 3   total_rooms         20640 non-null  float64
 4   total_bedrooms      20433 non-null  float64
 5   population          20640 non-null  float64
 6   households          20640 non-null  float64
 7   median_income       20640 non-null  float64
 8   median_house_value  20640 non-null  float64
 9   ocean_proximity     20640 non-null  object 
dtypes: float64(9), object(1)
memory usage: 1.6+ MB


OBSERVACIONES:
- Podemos observar de que la mayoria de nuestras columnas son del tipo float y solo la columna ocean proximity es de tipo objeto, por lo que deberemos cambiar el tipo de dato de las columnas. 
- Tambien observamos que en la columna total bedrooms hay una menor cantidad de datos por lo que deberian ser datos nulos.
- Finalmente el nombre de las columnas estan en ingles, por lo que para nuestra comodidad las cambiaremos a sus nombres en español.

In [47]:
#HACEMOS UN CONTEO DE LA CANTIDAD DE NULOS POR COLUMNAS
df_viviendas_california.isnull().sum()

longitude               0
latitude                0
housing_median_age      0
total_rooms             0
total_bedrooms        207
population              0
households              0
median_income           0
median_house_value      0
ocean_proximity         0
dtype: int64

Debido a que tenemos 207 nulos en la columna total_bedroooms procederemos a reemplazar estos valores con la media o la mediana. 


In [48]:
#UNA RAPIDA DESCRIPCION DE NUESTRO DATASET NOS MOSTRARA LA MEDIA Y LA MEDIANA DE NUESTRAS COLUMNAS.
df_viviendas_california.describe()

longitude      latitude  housing_median_age   total_rooms  \
count  20640.000000  20640.000000        20640.000000  20640.000000   
mean    -119.569704     35.631861           28.639486   2635.763081   
std        2.003532      2.135952           12.585558   2181.615252   
min     -124.350000     32.540000            1.000000      2.000000   
25%     -121.800000     33.930000           18.000000   1447.750000   
50%     -118.490000     34.260000           29.000000   2127.000000   
75%     -118.010000     37.710000           37.000000   3148.000000   
max     -114.310000     41.950000           52.000000  39320.000000   

       total_bedrooms    population    households  median_income  \
count    20433.000000  20640.000000  20640.000000   20640.000000   
mean       537.870553   1425.476744    499.539680       3.870671   
std        421.385070   1132.462122    382.329753       1.899822   
min          1.000000      3.000000      1.000000       0.499900   
25%        296.000000    787.000000    280.000000       2.563400   
50%        435.000000   1166.000000    409.000000       3.534800   
75%        647.000000   1725.000000    605.000000       4.743250   
max       6445.000000  35682.000000   6082.000000      15.000100   

       median_house_value  
count        20640.000000  
mean        206855.816909  
std         115395.615874  
min          14999.000000  
25%         119600.000000  
50%         179700.000000  
75%         264725.000000  
max         500001.000000

In [49]:
#VAMOS A VISUALIZAR NUESTRO DATASET
df_viviendas_california

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0        -122.23     37.88                41.0        880.0           129.0   
1        -122.22     37.86                21.0       7099.0          1106.0   
2        -122.24     37.85                52.0       1467.0           190.0   
3        -122.25     37.85                52.0       1274.0           235.0   
4        -122.25     37.85                52.0       1627.0           280.0   
...          ...       ...                 ...          ...             ...   
20635    -121.09     39.48                25.0       1665.0           374.0   
20636    -121.21     39.49                18.0        697.0           150.0   
20637    -121.22     39.43                17.0       2254.0           485.0   
20638    -121.32     39.43                18.0       1860.0           409.0   
20639    -121.24     39.37                16.0       2785.0           616.0   

       population  households  median_income  median_house_value  \
0           322.0       126.0         8.3252            452600.0   
1          2401.0      1138.0         8.3014            358500.0   
2           496.0       177.0         7.2574            352100.0   
3           558.0       219.0         5.6431            341300.0   
4           565.0       259.0         3.8462            342200.0   
...           ...         ...            ...                 ...   
20635       845.0       330.0         1.5603             78100.0   
20636       356.0       114.0         2.5568             77100.0   
20637      1007.0       433.0         1.7000             92300.0   
20638       741.0       349.0         1.8672             84700.0   
20639      1387.0       530.0         2.3886             89400.0   

      ocean_proximity  
0            NEAR BAY  
1            NEAR BAY  
2            NEAR BAY  
3            NEAR BAY  
4            NEAR BAY  
...               ...  
20635          INLAND  
20636          INLAND  
20637          INLAND  
20638          INLAND  
20639          INLAND  

[20640 rows x 10 columns]

In [50]:
#DEBIDO A QUE TENEMOS DATOS MUY GRANDES Y ALEJADOSDE LA MEDIA, PARA EVITAR UN ALTO SESGO PROCEDEROS A REEMPLAZAR 
#LOS VALORES NULOS CON LA MEDIANA
df_viviendas_california.fillna(435, inplace=True)

In [51]:
#A CONTINUACION PROCEDEREMOS A CAMBIAR EL NOMBRE DE LAS COLUMNAS
df_viviendas_california.rename(columns={'longitude': 'Longitud', 
                                'latitude': 'Latitud', 
                                'housing_median_age': 'Edad media de vivienda',
                                'total_rooms': 'Habitaciones en total',
                                'total_bedrooms': 'Dormitorios en total',
                                'population': 'Poblacion',
                                'households': 'Cantidad de hogares',
                                'median_income': 'Ingreso medio',
                                'median_house_value': 'Valor medio de viviendas',
                                'ocean_proximity': 'Proximidad al oceano'}, inplace=True)

In [52]:
df_viviendas_california.head(50)

Longitud  Latitud  Edad media de vivienda  Habitaciones en total  \
0    -122.23    37.88                    41.0                  880.0   
1    -122.22    37.86                    21.0                 7099.0   
2    -122.24    37.85                    52.0                 1467.0   
3    -122.25    37.85                    52.0                 1274.0   
4    -122.25    37.85                    52.0                 1627.0   
5    -122.25    37.85                    52.0                  919.0   
6    -122.25    37.84                    52.0                 2535.0   
7    -122.25    37.84                    52.0                 3104.0   
8    -122.26    37.84                    42.0                 2555.0   
9    -122.25    37.84                    52.0                 3549.0   
10   -122.26    37.85                    52.0                 2202.0   
11   -122.26    37.85                    52.0                 3503.0   
12   -122.26    37.85                    52.0                 2491.0   
13   -122.26    37.84                    52.0                  696.0   
14   -122.26    37.85                    52.0                 2643.0   
15   -122.26    37.85                    50.0                 1120.0   
16   -122.27    37.85                    52.0                 1966.0   
17   -122.27    37.85                    52.0                 1228.0   
18   -122.26    37.84                    50.0                 2239.0   
19   -122.27    37.84                    52.0                 1503.0   
20   -122.27    37.85                    40.0                  751.0   
21   -122.27    37.85                    42.0                 1639.0   
22   -122.27    37.84                    52.0                 2436.0   
23   -122.27    37.84                    52.0                 1688.0   
24   -122.27    37.84                    52.0                 2224.0   
25   -122.28    37.85                    41.0                  535.0   
26   -122.28    37.85                    49.0                 1130.0   
27   -122.28    37.85                    52.0                 1898.0   
28   -122.28    37.84                    50.0                 2082.0   
29   -122.28    37.84                    52.0                  729.0   
30   -122.28    37.84                    49.0                 1916.0   
31   -122.28    37.84                    52.0                 2153.0   
32   -122.27    37.84                    48.0                 1922.0   
33   -122.27    37.83                    49.0                 1655.0   
34   -122.27    37.83                    51.0                 2665.0   
35   -122.27    37.83                    49.0                 1215.0   
36   -122.27    37.83                    48.0                 1798.0   
37   -122.28    37.83                    52.0                 1511.0   
38   -122.26    37.83                    52.0                 1470.0   
39   -122.26    37.83                    52.0                 2432.0   
40   -122.26    37.83                    52.0                 1665.0   
41   -122.26    37.83                    51.0                  936.0   
42   -122.26    37.84                    49.0                  713.0   
43   -122.26    37.84                    52.0                  950.0   
44   -122.26    37.83                    52.0                 1443.0   
45   -122.26    37.83                    52.0                 1656.0   
46   -122.26    37.83                    50.0                 1125.0   
47   -122.27    37.82                    43.0                 1007.0   
48   -122.26    37.82                    40.0                  624.0   
49   -122.27    37.82                    40.0                  946.0   

    Dormitorios en total  Poblacion  Cantidad de hogares  Ingreso medio  \
0                  129.0      322.0                126.0         8.3252   
1                 1106.0     2401.0               1138.0         8.3014   
2                  190.0      496.0                177.0         7.2574   
3                  235.0      5

In [53]:
#AHORA PROCEDEMOS A CAMBIAR EL TIPO DE DATO DE LAS COLUMNAS
df_viviendas_california['Dormitorios en total'] = df_viviendas_california['Dormitorios en total'].astype('Int64')
df_viviendas_california['Edad media de vivienda'] = df_viviendas_california['Edad media de vivienda'].astype('Int64')
df_viviendas_california['Habitaciones en total'] = df_viviendas_california['Habitaciones en total'].astype('Int64')
df_viviendas_california['Cantidad de hogares'] = df_viviendas_california['Cantidad de hogares'].astype('Int64')

In [54]:
df_viviendas_california.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 10 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Longitud                  20640 non-null  float64
 1   Latitud                   20640 non-null  float64
 2   Edad media de vivienda    20640 non-null  Int64  
 3   Habitaciones en total     20640 non-null  Int64  
 4   Dormitorios en total      20640 non-null  Int64  
 5   Poblacion                 20640 non-null  float64
 6   Cantidad de hogares       20640 non-null  Int64  
 7   Ingreso medio             20640 non-null  float64
 8   Valor medio de viviendas  20640 non-null  float64
 9   Proximidad al oceano      20640 non-null  object 
dtypes: Int64(4), float64(5), object(1)
memory usage: 1.7+ MB


In [59]:
#AQUI PROCEDEMOS A CAMBIAR LAS FILAS DE LA COLUMNA PROXIMIDAD AL OCEANO A MINUSCULAS.
df_viviendas_california['Proximidad al oceano'] = df_viviendas_california['Proximidad al oceano'].str.capitalize()

In [58]:
df_viviendas_california

Longitud  Latitud  Edad media de vivienda  Habitaciones en total  \
0       -122.23    37.88                      41                    880   
1       -122.22    37.86                      21                   7099   
2       -122.24    37.85                      52                   1467   
3       -122.25    37.85                      52                   1274   
4       -122.25    37.85                      52                   1627   
...         ...      ...                     ...                    ...   
20635   -121.09    39.48                      25                   1665   
20636   -121.21    39.49                      18                    697   
20637   -121.22    39.43                      17                   2254   
20638   -121.32    39.43                      18                   1860   
20639   -121.24    39.37                      16                   2785   

       Dormitorios en total  Poblacion  Cantidad de hogares  Ingreso medio  \
0                       129      322.0                  126         8.3252   
1                      1106     2401.0                 1138         8.3014   
2                       190      496.0                  177         7.2574   
3                       235      558.0                  219         5.6431   
4                       280      565.0                  259         3.8462   
...                     ...        ...                  ...            ...   
20635                   374      845.0                  330         1.5603   
20636                   150      356.0                  114         2.5568   
20637                   485     1007.0                  433         1.7000   
20638                   409      741.0                  349         1.8672   
20639                   616     1387.0                  530         2.3886   

       Valor medio de viviendas Proximidad al oceano  
0                      452600.0             Near bay  
1                      358500.0             Near bay  
2                      352100.0             Near bay  
3                      341300.0             Near bay  
4                      342200.0             Near bay  
...                         ...                  ...  
20635                   78100.0               Inland  
20636                   77100.0               Inland  
20637                   92300.0               Inland  
20638                   84700.0               Inland  
20639                   89400.0               Inland  

[20640 rows x 10 columns]

In [60]:
#EXPORTAREMOS NUESTRO DATASET A UN ARCHIVO CSV
df_viviendas_california.to_csv('viviendas california.csv', index=False)
